### PR TITLE
All activities are ignored when excludeByCategory and limitByCategory are empty.

### PR DESCRIPTION
**Pull request by Sardis93**
[#157](https://github.com/ymcatwincities/openy_activity_finder/pull/157) introduced a code that hides all activities when both
**Limit by category** and **Exclude by category** fields of the Activity Finder 4 block are not set.

Steps to reproduce:

- Enable openy_prgf_af4_demo and ensure solr is indexed
- edit the AF4 paragraph make sure the **Limit by Category** and **Exclude by category** are both empty.
- Go to Activity Finder and select "Activity" step.
- Observe absence of table compeletely.

### CODE REVIEW

File: openy_af4_vue_app/src/components/steps/SelectActivities.vue [hunk 1]
```diff
@@ -113,7 +113,7 @@ export default {
         }
 
         const filteredValue = activityGroup.value.filter(item => {
-          let r = false
+          let r = true
           // Items must pass both tests, so we intentionally do not ELSE these.
           // If there are excludes, the item must NOT be excluded.
           if (this.excludeByCategory.length) {
```
Comment (anpolimus) [inline openy_af4_vue_app/src/components/steps/SelectActivities.vue:119]: This code now has no sense.
We are setting a variable to true at the beginning.
We are not changing it to false during the road and only returning that variable.
Let's double-check again if this PR changes anything.

**Review by anpolimus — CHANGES_REQUESTED** [review]

File: openy_af4_vue_app/src/components/steps/SelectActivities.vue (hunk 1 above)
Comment (Sardis93) [reply openy_af4_vue_app/src/components/steps/SelectActivities.vue:119]: So the logic behind this code is the following:

## With let r = true
By default we allow any item from `activityGroup` to be passed by setting
```
let r = true
```

If the value is present in the `excludeByCategory`, then the
```
r = !this.excludeByCategory.includes(item.value.toString())
```
will result in `false`, thus not adding the item into the resulting `filteredValue` array.

Then the value is checked again using `limitByCategory` array. 
If the value is missing in this array, then the
```
r = this.limitByCategory.includes(item.value.toString())
```
will return `false` once again, ensuring it is not in the resulting `filteredValue`.

## With let r = false
Now let's consider the case where:
```
let r = false
```
And where `excludeByCategory` & `limitByCategory` are both empty, i.e. the condition
```
if (this.excludeByCategory.length)
```
and 
```
if (this.limitByCategory.length)
```
are never passed.

The code never gets to checking
`r = !this.excludeByCategory.includes(item.value.toString())` and `r = this.limitByCategory.includes(item.value.toString())`
And it never gets a chance to change the `r` to be `true`.

Comment (anpolimus) [reply openy_af4_vue_app/src/components/steps/SelectActivities.vue:119]: Sorry, you were right.
I've missed this line of code:
`r = !this.excludeByCategory.includes(item.value.toString())`

**Review by anpolimus — APPROVED** [review]

**Review by anpolimus — APPROVED** [review]

**Comment by shelleygoetz** [comment]
Fix added to March release https://yusa.atlassian.net/browse/DS-414

**Pull request merged**